grading/load_config_json.cpp

```diff
@@ -20,6 +20,40 @@ void json_set_default(nlohmann::json &whole_config, std::string field, const T&
   }
 }
 
+uint64_t parse_file_size(std::string size_str) {
+  // Trim whitespaces
+  size_str.erase(remove_if(size_str.begin(), size_str.end(), isspace), size_str.end());
+
+  // Convert string to lowercase, note that it cannot handle UTF-8 characters
+  for (char& c : size_str)
+    c = std::tolower(c);
+
+  // Extract numeric file size, will cause exception if the number is larger than 64-bit
+  size_t unit_begin = 0;
+  uint64_t file_size = std::stoull(size_str, &unit_begin);
+  // check if it is 0 or contains invalid chars     ====whitelist====
+  if (file_size == 0 || size_str.find_first_not_of("01234567890bkmgti") == std::string::npos) {
```

Comment:
I haven't touched C++ from a long time, just here out of curiosity. Could you explain how this line is working? If this is doing a whitelist check then shouldn't there be `!= std::string::npos` ?

Reply:
Thanks for catching that.  Will fix in the next commit.